TKeazirian.HTTPServer.Tests/ResponseBuilderTests.cs

```diff
@@ -0,0 +1,48 @@
+using TKeazirian.HTTPServer.Tests.helpers;
+using Xunit;
+
+namespace TKeazirian.HTTPServer.Tests;
+
+public class ResponseBuilderTests
+{
+    [Fact]
+    public void ResponseWithOkStatusBuilt()
+    {
+        string responseStatus = Constants.Status200;
+
+        string responseHeaders = HelperFunctions.CreateTestResponseHeaders();
+
+        string? responseBody = "Hello world";
+
+        ResponseBuilder responseBuilder = new ResponseBuilder();
+
+        string expectedResponse =
+            responseStatus + Constants.NewLine +
+            responseHeaders + Constants.NewLine + Constants.NewLine +
+            responseBody;
+
+        string actualResponse = responseBuilder.BuildResponseForGet(responseStatus);
```

Comment:
with the builder pattern, you could theoretically do somthing like this:
```c-sharp
string testBody = "Hello world";
actualResponse =
    resonseBuilder.addStatus(200)
        .addHeader("Content-Type: text/plain)
        .addHeader("Content-Length: {body.Length}")
        .addBody(body)
        .build();
```

I suggest reading up on it more at: [Refactoring Guru: Design Patterns > Creational Patterns > Builder](https://refactoring.guru/design-patterns/builder)

Reply:
So maybe what I did was actually not the Builder pattern - I understand creating methods that can chain like this.  I'm not clear on what `build()` would look like if it doesn't take any parameters.  I trying to do something similar with the code in `Response.cs` but I may have gotten that wrong.

Reply:
With the builder pattern, You would need a `Response` object as well as a `ResponseBuilder` object.

The `ResponseBuilder` would allow the chaining of `ResponseBuilder.methods(), and then finally `ResponseBuilder.build()` method would return a new `Response` which would be configured all the things you chained.

Reply:
So maybe I did it slightly backward (and not quite right) - I have methods in my `Response` and then the `ResponseBuilder` just returns a `new Response` that puts them together...